### PR TITLE
Add support for nominal records

### PR DESCRIPTION
**Pull request by agocke**
Removes the requirement for a record to have a parameter list

### CODE REVIEW

File: src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs [hunk 1]
```diff
@@ -3063,8 +3066,12 @@ private void AddSynthesizedConstructorsIfNecessary(ArrayBuilder<Symbol> members,
                     switch (method.MethodKind)
                     {
                         case MethodKind.Constructor:
-                            hasInstanceConstructor = true;
-                            hasParameterlessInstanceConstructor = hasParameterlessInstanceConstructor || method.ParameterCount == 0;
+                            // Ignore the record copy constructor
+                            if (!(method is SynthesizedRecordCopyCtor))
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs:3070]: Can we check 'method.IsImplicitlyDeclared' here instead?

Comment (agocke) [reply src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs:3070]: That's true for all SynthesizedInstanceMethodSymbols, including the record primary constructor, unfortunately

File: src/Compilers/CSharp/Test/Symbol/Symbols/Source/RecordTests.cs [hunk 2]
```diff
@@ -812,5 +815,125 @@ .maxstack  2
...
+            Assert.Equal(v1.VisualizeIL("C.Equals(C)"), v2.VisualizeIL("C.Equals(C)"));
+            Assert.Equal(v1.VisualizeIL("C.Equals(object)"), v2.VisualizeIL("C.Equals(object)"));
+        }
+
+        [Fact]
+        public void NominalRecordMembers()
+        {
+            var comp = CreateCompilation(@"
+#nullable enable
+data class C
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Symbol/Symbols/Source/RecordTests.cs:915]: Please add a test for `data class C()`. From your PR description, I'm not sure what is expected (allowed or not?)

Comment (agocke) [reply src/Compilers/CSharp/Test/Symbol/Symbols/Source/RecordTests.cs:915]: Right the absence of a record parameter list is not the same as an empty record parameter list. Right now it's an error.

File: src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs [hunk 3]
```diff
@@ -2927,36 +2927,39 @@ private void AddSynthesizedRecordMembersIfNecessary(MembersAndInitializersBuilde
...
-
             // PROTOTYPE: need to check base members as well
             var memberSignatures = s_duplicateMemberSignatureDictionary.Allocate();
             foreach (var member in members)
             {
                 memberSignatures.Add(member, member);
             }
 
-            var ctor = addCtor(paramList);
+            // Positional record
```
Comment (jcouv) [inline src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs:2943]: >Positional record [](start = 15, length = 17)

nit: positional members?

**Pull request merged**